vscode/src/debugger/session.ts

```diff
@@ -874,4 +813,57 @@ export class QscDebugSession extends LoggingDebugSession {
     );
     this.sendEvent(evt);
   }
+
+  /**
+   * Attempts to find the Source in the current session and returns the
+   * TextDocument if it exists.
+   *
+   * This method *may* return a valid result even when the requested
+   * path does not belong in the current program (e.g. another Q# file
+   * in the workspace).
+   */
+  async tryLoadSource(source: DebugProtocol.Source) {
+    if (!source.path) {
+      return;
+    }
+
+    const uri = this.asUri(source.path);
+    if (!uri) {
+      return;
+    }
+
+    try {
+      const doc = await vscode.workspace.openTextDocument(uri);
+      if (!isQsharpDocument(doc)) {
+        return;
+      }
+      return doc;
+    } catch (e) {
+      log.trace(`Failed to open ${uri}: ${e}`);
+    }
+  }
+
+  /**
+   * Attemps to resolve a DebugProtocol.Source.path to a URI.
+   *
+   * This method will attempt to undo the lossy conversion
+   * from URI to path that the VS Code debug adapter client performs.
+   * See: https://github.com/microsoft/vscode/blob/3246d63177e1e5ae211029e7ab0021c33342a3c7/src/vs/workbench/contrib/debug/common/debugSource.ts#L90
+   *
+   * If the path was originally constructed from a file:// URI, it's a
+   * filesystem path. We use our knownPaths map to resolve it back to a URI.
+   * Filesystem paths we don't know about *won't* be resolved, and that's ok.
```

Comment:
I've been looking at the comments here and on the `knownPaths` member for a bit and still can't fully grok why we need this or what problem it solves. Can you try to clarify a bit? Thanks.